File: app/src/main/java/intellij/class5/objetos/exercise1/java/Conta.java
package intellij.class5.objetos.exercise1.java;

public class Conta {
    private String numeroDaConta;
    private double saldo = 0.0;
    private Cliente titular;

    public Conta() {
    }

    public Conta(String numeroDaConta, Cliente titular) {
        setNumeroDaConta(numeroDaConta);
        setSaldo(saldo);
        setTitular(titular);
    }

    public Conta(String numeroDaConta, double saldo, Cliente titular) {
        setNumeroDaConta(numeroDaConta);
        setSaldo(saldo);
        setTitular(titular);
    }

    public double makeDeposit(double value) {
        setSaldo(getSaldo() + value);
        System.out.println("Depósito. \nSaldo: " + getSaldo());
        return getSaldo();
    }

    public double makeWithdraw(double value) {
        if (value > getSaldo()) {
            System.out.println("Saldo Insuficiente.");
        } else {
            setSaldo(getSaldo() - value);
            System.out.println("Saque. \nSaldo: " + getSaldo());
        }
        return getSaldo();
    }

    public String getNumeroDaConta() {
        return numeroDaConta;
    }

    public void setNumeroDaConta(String numeroDaConta) {
        this.numeroDaConta = numeroDaConta;
    }

    public double getSaldo() {
        return saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public Cliente getTitular() {
        return titular;
    }

    public void setTitular(Cliente titular) {
        this.titular = titular;
    }
}

File: app/src/main/java/intellij/class5/objetos/exercise1/kotlin/Conta.kt
package intellij.class5.objetos.exercise1.kotlin

data class Conta(
    var numeroDaConta: String,
    var saldo: Double = 0.0,
    var titular: Cliente
) {

    //Segundo Construtor (opcional)
    constructor(numeroDaConta: String, titular: Cliente) : this(numeroDaConta, 0.0, titular)

    fun makeDeposit(value: Double): Double {
        saldo += value
        println("Depósito.\nSaldo: $saldo")
        return saldo
    }

    fun makeWithdraw(value: Double): Double {
        if (value > saldo) {
            println("Saldo Insuficiente")
        } else {
            saldo -= value
            println("Saque.\nSaldo: $saldo")
        }
        return saldo
    }
}